### PR TITLE
ios 13 semantic and dynamic color support

### PR DESCRIPTION
**Pull request by tom-un**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

#### Description of changes

This change adds semantic named color support to iOS, in the same way that macOS react-native supports semantic and dynamic colors.

The changes in this branch were discussed in the following thread regarding dark mode implementation proposals: https://github.com/react-native-community/discussions-and-proposals/issues/126.   It will be committed to this Microsoft fork now for support need in Microsoft products.   This support will be contributed to the facebook repo when a community consensus is reached.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/105)

### CODE REVIEW

**Comment by tom-un** [comment]
> What about support for similar dark/high contrast theming in Android Q? The platform checks in here make it unfriendly to Android, Windows, and other platforms.

My thought was that the `{ semantic: "some-name" }` syntax would work for all platforms, but of course it would require a change to the android native code to support it which I'm not equipped to implement.  Until such support exists, I thought it prudent to have the platform checks in the JS so that the JS would behave exactly the same way it did without this change on non apple platforms.

I had hoped that some kind of platform agnostic semantic names could be developed that span all platforms.   But from my own experience writing JS code that is common between ios and mac, even between two *Apple* platforms the semantic colors names are not at all aligned.  NSColor names (macos) and UIColor names (ios) are quite different.   And in our scenarios, react-native features are embedded within larger, purely native apps.  The purely native parts of the app are using semantic colors and materials, so its critical that the react-native embedded experiences use the same palette of semantic color names so that the overall experience is seamless.  For instance, in this example from Mac Outlook the main window is native and the popup card is react-native:

image

Both the native experience and the react-native experience are using NSColor semantic colors so they both respond seamlessly to theme changes and should be robust against any changes Apple makes to these colors in future releases.

I considered naming the keys something platform specific, like `{ semantic-ios: "" }` and `{ semantic-macos: "" }`, but it seemed gratuitous.

File: React/Base/RCTConvert.m [hunk 1]
```diff
@@ -614,11 +616,114 @@ + (type)type:(id)json                                       \
       @"systemPurpleColor": @{}, // 10_10
       @"systemRedColor": @{},    // 10_10
       @"systemYellowColor": @{}, // 10_10
+#else // TARGET_OS_IOS
+      // https://developer.apple.com/documentation/uikit/uicolor/ui_element_colors
+      // Label Colors
+      @"labelColor": @{ // 13
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Without the context of 10_10 above, it was very unclear what the "13" means. Probably doesn't hurt to precede with iOS13 or something like that instead

File: React/Base/RCTConvert.m [hunk 2]
```diff
@@ -614,11 +616,114 @@ + (type)type:(id)json                                       \
...
+      },
+      @"secondarySystemGroupedBackgroundColor": @{ // 13
+        RCTFallbackARGB: @(0xFFffffff)
+      },
+      @"tertiarySystemGroupedBackgroundColor": @{ // 13
+        RCTFallbackARGB: @(0xFFf2f2f7)
+      },
+      // Separator Colors
+      @"separatorColor": @{ // 13
+        RCTFallbackARGB: @(0x493c3c43)
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Maybe I missed the comment somewhere, but all these hex values came from where?

File: React/Base/RCTConvert.m [hunk 3]
```diff
@@ -709,35 +819,50 @@ + (UIColor *)UIColor:(id)json
...
     return [UIColor colorWithRed:r green:g blue:b alpha:a];
-#if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
+// [TODO(macOS ISS#2323203)
   } else if ([json isKindOfClass:[NSDictionary class]]) {
     NSDictionary *dictionary = json;
     id value = nil;
     if ((value = [dictionary objectForKey:@"semantic"])) {
       NSString *semanticName = value;
-      NSColor *color = RCTColorFromSemanticColorName(semanticName);
+      UIColor *color = RCTColorFromSemanticColorName(semanticName);
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Is UIColor macro'd to NSColor or something? I still think that pattern is pretty confusing. In my opinion, we should at least create an RCTUIColor that is NSColor for Mac and UIColor on iOS. Then it's at least apparent we aren't attempting to manipulate an iOS object on Mac

Comment (HeyImChris) [reply React/Base/RCTConvert.m]: Or does this just not run on Mac? I see you removed the if def, but I'm assuming we don't want to lose Mac functionality

File: React/Base/RCTConvert.m [hunk 4]
```diff
@@ -709,35 +819,50 @@ + (UIColor *)UIColor:(id)json
...
       if (color == nil) {
-        RCTLogConvertError(json, [@"an NSColor.  Expected one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
+        RCTLogConvertError(json, [@"an UIColor.  Expected one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: "a UIColor" but also, this doesn't run on Mac?

File: React/Base/RCTConvert.m [hunk 5]
```diff
@@ -709,35 +819,50 @@ + (UIColor *)UIColor:(id)json
...
+#else
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+        if (@available(iOS 13.0, *)) {
+          UIColor *color = [UIColor colorWithDynamicProvider:^UIColor * _Nonnull(UITraitCollection * _Nonnull collection) {
+            return collection.userInterfaceStyle == UIUserInterfaceStyleLight ? lightColor : darkColor;
+          }];
+          return color;
+        } else {
+#endif
+          return lightColor;
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: indentation doesn't look right

File: React/Base/RCTConvert.m [hunk 6]
```diff
@@ -709,35 +819,50 @@ + (UIColor *)UIColor:(id)json
...
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+        }
+#endif
+#endif
       } else {
-        RCTLogConvertError(json, @"an NSColor. Expected a mac dynamic appearance aware color.");
+        RCTLogConvertError(json, @"an UIColor. Expected a mac dynamic appearance aware color.");
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: "a". I'll stop commenting this but it happens more throughout this review

File: React/Base/RCTConvert.m (hunk 3 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: It is that UIColor is defined to NSColor via RCTUIKit.h.   I'll investigate refactoring stuff to RCTUIColor

File: React/Base/RCTConvert.m (hunk 2 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: They were generated some logging breakpoints I added to a unit test.  They are the RGBA values resolved from the semantic colors on iOS 13 in light mode.

File: React/Base/RCTConvert.m (hunk 4 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: It is an NSColor on mac, UIColor on iOS via RCTUIKit.h

File: React/Base/RCTConvert.m (hunk 5 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: It aligns with the if (@available()) {

File: React/Base/RCTConvert.m (hunk 3 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: Done

File: React/Base/RCTConvert.m (hunk 2 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: Adding comment

File: Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m [hunk 7]
```diff
@@ -38,7 +38,7 @@ - (instancetype)initWithBridge:(RCTBridge *)bridge
     _backedTextInputView.scrollEnabled = YES;
 #else // [TODO(macOS ISS#2323203)
     _scrollView = [[RCTUIScrollView alloc] initWithFrame:self.bounds]; // TODO(macOS ISS#3536887)
-    _scrollView.backgroundColor = [UIColor clearColor];
+    _scrollView.backgroundColor = [RCTUIColor clearColor];
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m [hunk 8]
```diff
@@ -104,17 +104,17 @@ - (void)setPlaceholderColor:(UIColor *)placeholderColor
 }
 
 #if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
-- (void)setSelectionColor:(UIColor *)selectionColor
+- (void)setSelectionColor:(RCTUIColor *)selectionColor
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Multiline/RCTUITextView.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m [hunk 9]
```diff
@@ -104,17 +104,17 @@ - (void)setPlaceholderColor:(UIColor *)placeholderColor
...
 {
   NSMutableDictionary *selectTextAttributes = self.selectedTextAttributes.mutableCopy;
   selectTextAttributes[NSBackgroundColorAttributeName] = selectionColor ?: [NSColor selectedControlColor];
   self.selectedTextAttributes = selectTextAttributes.copy;
   self.insertionPointColor = self.selectionColor ?: [NSColor selectedControlColor];
 }
 
-- (UIColor*)selectionColor
+- (RCTUIColor*)selectionColor
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Multiline/RCTUITextView.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m [hunk 10]
```diff
@@ -104,17 +104,17 @@ - (void)setPlaceholderColor:(UIColor *)placeholderColor
...
 {
-  return (UIColor*)self.selectedTextAttributes[NSBackgroundColorAttributeName];
+  return (RCTUIColor*)self.selectedTextAttributes[NSBackgroundColorAttributeName];
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Multiline/RCTUITextView.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Singleline/RCTUITextField.h [hunk 11]
```diff
@@ -42,7 +42,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, copy, nullable) NSString *text;
 @property (nonatomic, getter=isAutomaticTextReplacementEnabled) BOOL automaticTextReplacementEnabled;
 @property (nonatomic, getter=isAutomaticSpellingCorrectionEnabled) BOOL automaticSpellingCorrectionEnabled;
-@property (nonatomic, strong, nullable) UIColor *selectionColor;
+@property (nonatomic, strong, nullable) RCTUIColor *selectionColor;
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.h]: Missing the TODO comment

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m [hunk 12]
```diff
@@ -18,7 +18,7 @@ @interface RCTUITextFieldCell : NSTextFieldCell
 @property (nonatomic, assign) UIEdgeInsets textContainerInset;
 @property (nonatomic, getter=isAutomaticTextReplacementEnabled) BOOL automaticTextReplacementEnabled;
 @property (nonatomic, getter=isAutomaticSpellingCorrectionEnabled) BOOL automaticSpellingCorrectionEnabled;
-@property (nonatomic, strong, nullable) UIColor *selectionColor;
+@property (nonatomic, strong, nullable) RCTUIColor *selectionColor;
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m [hunk 13]
```diff
@@ -65,7 +65,7 @@ - (NSText *)setUpFieldEditorAttributes:(NSText *)textObj
   NSMutableDictionary *selectTextAttributes = fieldEditor.selectedTextAttributes.mutableCopy;
   selectTextAttributes[NSBackgroundColorAttributeName] = self.selectionColor ?: [NSColor selectedControlColor];
 	fieldEditor.selectedTextAttributes = selectTextAttributes;
-  fieldEditor.insertionPointColor = self.selectionColor ?: [NSColor selectedControlColor];
+  fieldEditor.insertionPointColor = self.selectionColor ?: [RCTUIColor selectedControlColor];
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m [hunk 14]
```diff
@@ -85,10 +85,10 @@ @implementation RCTUITextField {
   return [UIFont systemFontOfSize:17];
 }
 
-static UIColor *defaultPlaceholderTextColor()
+static RCTUIColor *defaultPlaceholderTextColor()
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m [hunk 15]
```diff
@@ -85,10 +85,10 @@ @implementation RCTUITextField {
...
 {
   // Default placeholder color from UITextField.
-  return [UIColor colorWithRed:0 green:0 blue:0.0980392 alpha:0.22];
+  return [RCTUIColor colorWithRed:0 green:0 blue:0.0980392 alpha:0.22];
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/Singleline/RCTUITextField.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 16]
```diff
@@ -740,21 +740,21 @@ + (type)type:(id)json                                       \
   return colorMap;
 }
 
-/** Returns an UIColor based on a semantic color name.
+/** Returns a UIColor based on a semantic color name.
  *  Returns nil if the semantic color name is invalid.
  */
-static UIColor *RCTColorFromSemanticColorName(NSString *semanticColorName)
+static RCTUIColor *RCTColorFromSemanticColorName(NSString *semanticColorName)
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 17]
```diff
@@ -740,21 +740,21 @@ + (type)type:(id)json                                       \
...
 {
   NSDictionary<NSString *, NSDictionary *> *colorMap = RCTSemanticColorsMap();
-  UIColor *color = nil;
+  RCTUIColor *color = nil;
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 18]
```diff
@@ -740,21 +740,21 @@ + (type)type:(id)json                                       \
...
   NSDictionary<NSString *, id> *colorInfo = colorMap[semanticColorName];
   if (colorInfo) {
     NSString *semanticColorSelector = colorInfo[RCTSelector];
     if (semanticColorSelector == nil) {
       semanticColorSelector = semanticColorName;
     }
     SEL selector = NSSelectorFromString(semanticColorSelector);
-    if (![UIColor respondsToSelector:selector]) {
+    if (![RCTUIColor respondsToSelector:selector]) {
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 19]
```diff
@@ -763,12 +763,12 @@ + (type)type:(id)json                                       \
       semanticColorSelector = colorInfo[RCTFallback];
       selector = NSSelectorFromString(semanticColorSelector);
     }
-    RCTAssert ([UIColor respondsToSelector:selector], @"UIColor does not respond to a semantic color selector.");
-    Class klass = [UIColor class];
+    RCTAssert ([RCTUIColor respondsToSelector:selector], @"RCTUIColor does not respond to a semantic color selector.");
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 20]
```diff
@@ -763,12 +763,12 @@ + (type)type:(id)json                                       \
...
+    Class klass = [RCTUIColor class];
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 21]
```diff
@@ -763,12 +763,12 @@ + (type)type:(id)json                                       \
...
     IMP imp = [klass methodForSelector:selector];
     id (*getSemanticColorObject)(id, SEL) = (void *)imp;
     id colorObject = getSemanticColorObject(klass, selector);
-    if ([colorObject isKindOfClass:[UIColor class]]) {
+    if ([colorObject isKindOfClass:[RCTUIColor class]]) {
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 22]
```diff
@@ -800,42 +800,42 @@ + (type)type:(id)json                                       \
...
-    return [UIColor colorWithRed:r green:g blue:b alpha:a];
+    return [RCTUIColor colorWithRed:r green:g blue:b alpha:a]; // TODO(OSS Candidate ISS#2710739)
 // [TODO(macOS ISS#2323203)
   } else if ([json isKindOfClass:[NSDictionary class]]) {
     NSDictionary *dictionary = json;
     id value = nil;
     if ((value = [dictionary objectForKey:@"semantic"])) {
       NSString *semanticName = value;
-      UIColor *color = RCTColorFromSemanticColorName(semanticName);
+      RCTUIColor *color = RCTColorFromSemanticColorName(semanticName);
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 23]
```diff
@@ -800,42 +800,42 @@ + (type)type:(id)json                                       \
...
       if (color == nil) {
-        RCTLogConvertError(json, [@"an UIColor.  Expected one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
+        RCTLogConvertError(json, [@"a UIColor.  Expected one of the following values: " stringByAppendingString:RCTSemanticColorNames()]);
       }
       return color;
     } else if ((value = [dictionary objectForKey:@"dynamic"])) {
       NSDictionary *appearances = value;
       id light = [appearances objectForKey:@"light"];
-      UIColor *lightColor = [RCTConvert UIColor:light];
+      RCTUIColor *lightColor = [RCTConvert UIColor:light];
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Base/RCTConvert.m [hunk 24]
```diff
@@ -800,42 +800,42 @@ + (type)type:(id)json                                       \
...
       id dark = [appearances objectForKey:@"dark"];
-      UIColor *darkColor = [RCTConvert UIColor:dark];
+      RCTUIColor *darkColor = [RCTConvert UIColor:dark];
```
Comment (HeyImChris) [inline React/Base/RCTConvert.m]: Missing the TODO comment

File: React/Views/RCTActivityIndicatorView.m [hunk 25]
```diff
@@ -66,7 +66,7 @@ - (void)setActivityIndicatorViewStyle:(UIActivityIndicatorViewStyle)activityIndi
   }
 }
 
-- (void)setColor: (UIColor*)color
+- (void)setColor: (RCTUIColor*)color
```
Comment (HeyImChris) [inline React/Views/RCTActivityIndicatorView.m]: Missing the TODO comment

File: React/Views/RCTProgressView.h [hunk 26]
```diff
@@ -17,8 +17,8 @@
 #endif
 
 #if TARGET_OS_OSX
-@property (nonatomic, strong, nullable) UIColor *progressTintColor;
-@property (nonatomic, strong, nullable) UIColor *trackTintColor;
+@property (nonatomic, strong, nullable) RCTUIColor *progressTintColor;
+@property (nonatomic, strong, nullable) RCTUIColor *trackTintColor;
```
Comment (HeyImChris) [inline React/Views/RCTProgressView.h]: Missing the TODO comment

File: React/Views/RCTProgressView.m [hunk 27]
```diff
@@ -19,7 +19,7 @@ - (instancetype)initWithFrame:(CGRect)frame
     self.indeterminate = NO;
 
     // Default track color from NSProgressIndicator.
-    self.trackTintColor = [UIColor colorWithRed: 237/255.0 green:237/255.0 blue:237/255.0 alpha:1.0];
+    self.trackTintColor = [RCTUIColor colorWithRed: 237/255.0 green:237/255.0 blue:237/255.0 alpha:1.0];
```
Comment (HeyImChris) [inline React/Views/RCTProgressView.m]: Missing the TODO comment

File: Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m (hunk 7 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m]: this change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m (hunk 8 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m (hunk 9 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m (hunk 10 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Singleline/RCTUITextField.h (hunk 11 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.h]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m (hunk 12 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m (hunk 13 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m (hunk 14 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: Libraries/Text/TextInput/Singleline/RCTUITextField.m (hunk 15 above)
Comment (tom-un) [reply Libraries/Text/TextInput/Singleline/RCTUITextField.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 16 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 17 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 18 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 19 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 21 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 22 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 23 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Base/RCTConvert.m (hunk 24 above)
Comment (tom-un) [reply React/Base/RCTConvert.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Views/RCTActivityIndicatorView.m (hunk 25 above)
Comment (tom-un) [reply React/Views/RCTActivityIndicatorView.m]: This change is already inside a ` // [TODO(macOS ISS#2323203)` ... ` // ]TODO(macOS ISS#2323203)` block.

File: React/Views/RCTProgressView.h (hunk 26 above)
Comment (tom-un) [reply React/Views/RCTProgressView.h]: Entire file is `// TODO(macOS ISS#2323203)`

File: React/Views/RCTProgressView.m (hunk 27 above)
Comment (tom-un) [reply React/Views/RCTProgressView.m]: Entire file is ` // TODO(macOS ISS#2323203)`

File: Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m (hunk 7 above)
Comment (HeyImChris) [reply Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m]: The line above it is marked specifically with a with a todo, makes it confusing if we're inconsistent

Comment (tom-un) [reply Libraries/Text/TextInput/Multiline/RCTMultilineTextInputView.m]: There *are* inconsistencies.   The comment on line 40 for RCTUIScrollView is not necessary as this is inside a `[` ... `]` block

**Pull request merged**